docs/workflow-job-submission.md

```diff
@@ -0,0 +1,180 @@
+that can be set.
+
+#### Reuse in job_cluster submission method
+
+If the following values are defined in `config.workflow_job_config`, they will be used even if
+the model uses the job_cluster submission method. For example, you can define a job_cluster model
+to send an email notification on failure.
+
+- grants
+- email_notifications
+- webhook_notifications
+- notification_settings
+- timeout_seconds
+- health
+- environments
+
+#### Workflow name
+
+The name of the workflow must be unique unless you also define an existing job id. By default,
+dbt will generate a name based on the catalog, schema, and model identifier.
+
+#### Clusters
+
+- If defined, dbt will re-use the `config.job_cluster_config` to define a job cluster for the workflow tasks.
+- If `config.workflow_job_config.existing_cluster_id` is defined, dbt will use that cluster
+- Similarly, you can define a reusable job cluster for the workflow and tell the task to use that 
+- If none of those are in the configuration, the task cluster will be serverless
+
+```yaml
+# Reusable job cluster config example
+
+config:
+  workflow_job_config:
+    additional_task_settings: {
+      task_key: 'task_a',
+      job_cluster_key: 'cluster_a',
+    }
+    post_hook_tasks: [{
+      "depends_on": [{ "task_key": "task_a" }],
+      "task_key": 'OPTIMIZE_AND_VACUUM',
+      job_cluster_key: 'cluster_a',
+      "notebook_task": {
+        "notebook_path": "/OPTIMIZE_AND_VACUUM",
+        "source": "WORKSPACE",
+      },
+    }]
+    job_clusters: [{
+      job_cluster_key: 'cluster_a',
+      new_cluster: {
+        spark_version: "{{ var('dbr_versions')['lts_v14'] }}",
+        node_type_id: "{{ var('cluster_node_types')['large_job'] }}",
+        runtime_engine: "{{ var('job_cluster_defaults.runtime_engine') }}",
+        autoscale: {
+          "min_workers": 1,
+          "max_workers": 2
+        },
+      }
+    }]
+```
+
+#### Grants
+
+You might want to give certain users or teams access to run your workflows outside of
+dbt in an ad hoc way. You can define those permissions in the `workflow_job_config.grants`.
+The owner will always be the user or service principal creating the workflows.
+
+These grants will also be applied to one-time run models using the `job_cluster` submission 
+method.
+
+The dbt rules correspond with the following Databricks permissions:
+
+- view: `CAN_VIEW`
+- run: `CAN_MANAGE_RUN`
+- manage: `CAN_MANAGE`
+
+```
+grants:
+  view: [
+    {"group_name": "marketing-team"},
+  ]
+  run: [
+    {"user_name": "alighodsi@databricks.com"}
+  ]
+  manage: []
+```
+
+#### Post hooks
+
+It is possible to add in python hooks by using the `config.workflow_job_config.post_hook_tasks` 
+attribute. If there is a cluster defined in `config.job_cluster_config`, that will get pulled 
+into the hook as a new cluster. For reusable job clusters or an existing job cluster, you will 
+have to define that in the post hook task definition manually.
```

Comment:
@benc-db - I've been going back and forth on this behavior. On the one hand, defaulting to separate job clusters might not be ideal if their definitions are the same. On the other hand, this code seems complex enough as it is and doesn't need _too_ much dbt automagic. If you have an opinion, let me know and I can make any changes, otherwise I'll probably just leave it.

Reply:
Actually, I'm going to scratch that and make clusters be defined for the post hooks. Otherwise you can't use a job cluster for the main notebook and serverless for a followup. And it will take out some magic.